### DISCUSSION TITLE
Anything similar to Dark Reader, but for the Desktop?

### DISCUSSION THREAD

**Discussion by chriscz** [darkreader/darkreader#8287] [Q&A]
Are you aware of any software package out there that does what Dark Reader does, but for the desktop? I run Ubuntu Linux with KDE and would love if there was a way to make some of those pesky GTK applications dark without necessarily using a theme.

**Comment by Gusted** [comment]
Search for "Dark plasma themes" they are the closest you can get for KDE, there is no universal fix like Dark Reader for DE's.

**Reply by chriscz** [reply]
Thanks Gusted, I already run plasma Dark, was just curious if there's anything out there. I wonder what level of hackery it would take to get something like Dark Reader working. Maybe one could hook into the frame-buffer, run some form of box-detection and change the colours of the pixels, text might be challenging though. 

I'll leave this open for a bit longer to see if anyone else responds in the next week :)

**Comment by KamilDev** [comment]
I came here from google, wondering the same thing (but on Windows). I believe there exists system patchers such as [Rectify11](https://github.com/Rectify11/Installer) which patch all apps to have dark mode, however:
- A bit dangerous patching system files
- I don't think it allows a per-app dark mode toggle. It's crucial to be able to easily disable dark mode if it doesn't work correctly in a specific app.

I'm assuming dark mode patches will have to be individually built for each kind of UI framework (e.g. Win32, WPF). I came across a [Unity dark mode mod](https://github.com/0x7c13/UnityEditor-DarkMode?tab=readme-ov-file#how-it-works) which changes the Windows native system menus in Unity to be in dark mode. It utilizes [ReaperThemeHackDll](https://github.com/jjYBdx4IL/ReaperThemeHackDll).


I was day dreaming of how cool it'd be to have something like Dark Reader for desktop apps, with the ability to toggle dark mode per-app. I'm not sure if something like this already exists, but please let me know if you find anything.

**Reply by Myshor** [reply]
> I came here from google, wondering the same thing (but on Windows).

Everything depends what version of Windows you want to make fully dark.

Most secure (I mean not breaking too much in Windows) is to use Niivu themes.
I have Windows 11 24H2 with https://github.com/niivu/Windows-11-themes/tree/main/Windows%2011%20redux

For patching I use 3 different software:

1. [Mica for Everyone](https://github.com/MicaForEveryone/MicaForEveryone) to have main window titlebar dark for all apps.
2. "themetool" - [SecureUxTheme](https://github.com/namazso/SecureUxTheme) - to patch windows to use 3rd Party themes.
3. OldNewExplorer (can be found using Google Search) - this fixes some old window color issues for Example in Control Panel.

P.S.
I use "Themetool" with this settings so even the wallpapers, cursors, icons and sounds stays untouched.
![obraz](https://github.com/user-attachments/assets/b0b80457-b6b1-45a7-b84b-232cf35c8f7a)

P.P.S.
If you still use windows 10 there is some themes on Niivu github for Windows 10 too. If I remember right I was using https://github.com/niivu/Windows-10-themes/tree/main/BIB3 with WIN scheme to have it as much similar as possible to default Windows 10.

**Reply by KamilDev** [reply]
Thanks for your information. I use Windows 11.

I definitely think I will try out [SecureUxTheme](https://github.com/namazso/SecureUxTheme) as it seems to have safety in mind, and [Mica for Everyone](https://github.com/MicaForEveryone/MicaForEveryone). However, I assume I shouldn't use OldNewExplorer as I want to keep the exact same Windows 11 Explorer?

Is it possible to find detailed steps to everything needed to setup to make as much applications in dark mode as possible?

I assume I should download [Windows 11 redux](https://github.com/niivu/Windows-11-themes/tree/main/Windows%2011%20redux) and use SecureUxTheme to apply it? Looking at your screenshot, I should apply the `Aero night.msstyles` theme with all those ignore checkboxes checked to avoid unintended extra visual changes.

As I'm only interested in making applications/windows that don't have dark mode to be in dark mode, and no other visual changes apart from that, will this achieve this goal? Will it make my Windows look any different apart from just making everything in dark mode?

**Reply by Myshor** [reply]
> I assume I shouldn't use OldNewExplorer as I want to keep the exact same Windows 11 Explorer?

I just don't change any default settings just install it. Then it keeps Explorer app untouched but patch some white places like I mentioned (Control Panel for example).

For the detailed steps:

1. Download SecureUxTheme
2. Run as Administrator.
3. Check all checkboxes here and install:
![obraz](https://github.com/user-attachments/assets/683e236a-2cb9-4bea-9492-20745cf775be)
4.  Download OldNewExplorer.
5. Run as Admin - do not touch any options. Just Install.
![obraz](https://github.com/user-attachments/assets/4f7f643c-0739-4513-91ed-d69ba6a882f5)
6.  Download MicaForEveryone. Install. Do not run if install ask about it. (I just don't remember if it asks or not).
7. Run Mica as Administrator!
8. Set the General Options like this:
![obraz](https://github.com/user-attachments/assets/b739b45f-d4ec-4730-98b7-e79341c2ffa7)
9. Set the Global rule like this:
![obraz](https://github.com/user-attachments/assets/be13b447-ea5f-44c2-93d9-92be0fc973d8)
The rest I leave utouched.
10. Now the main theme part. Download the Windows 11 Redux theme.
11. You can extract all themes from `Windows 11 themes` into `C:\Windows\Resources\Themes\` dir. (If your Windows directory is `C:\Windows\` if other then just find `\Resources\Themes\` in your windows directory).
12. This is optional but makes some cleaning and removing all unnecessary.
- Remove all Windows 11 themes related to Redux pack except `Windows 11 aero - Night.theme`.
- Go into `Windows 11 redux` subdirectory and remove all msstyles except `Aero night.msstyles`
13. Now we go back to SecureUxTheme.
14. Run as Administrator.
15. Go step by step:
- Select `Windows 11 aero - Night`
- Click "Patch" button
- Select all checkboxes to leave your current desktop settings untouched.
- Apply theme to have the full dark theme enabled.
![obraz](https://github.com/user-attachments/assets/96355c81-cd85-40be-ac1f-ad1ed1bfebbe)

Now all apps which doesn't use hard coded white color as background will start to work as dark.

Known issues:
1. Microsoft Excel cells uses window background color so it will change into dark after applying 3rd party themes. Workaround: use other Office suite (Libre Office for example) or change your theme to default windows used before this 3rd party theme.
(No screen here as I do not work with Microsoft Office).
2. Some apps (in my experience Brother scanning software for example) have hard coded white background and some places where gets text color for theme. In that situation that text will not be black but light gray. You need to live with it or change your theme to default windows used before this 3rd party theme when using this kind of soft.
![obraz](https://github.com/user-attachments/assets/3d722178-6a96-4f70-89c1-c1bee4b3ed85)

**Reply by KamilDev** [reply]
When some apps do not work well with the custom dark theme, how easy is it to switch back to regular Windows theme temporarily while using the app? Does it require restarting the computer?

**Reply by Myshor** [reply]
No... you just switch the theme in Windows Settings or using SecureUxTheme. After all previous patching steps done none restarts are needed.

One more tip... you can select third party theme from Windows Settings but it's not recommended. But when you have it already selected using SecureUxTheme you can set your desired background for it... After changing background you can save modified theme and then select it from Windows Settings without using SecureUxTheme at all. 😉

I mentioned it as I use 3rd Party Dark theme with Microsoft Spotlight changing wallpapers. 😁

**Reply by slow-played-zunes** [reply]
> When some apps do not work well with the custom dark theme, how easy is it to switch back to regular Windows theme temporarily while using the app? Does it require restarting the computer?

You can instead use Windows high-contrast themes and set the colors to whateer you want, even works great in Edge, but there is not a way I can find to toggle it on or off in Edge, so if you use high-contrast windows theme, edge forces the pages to use those custom colors too.

**Comment by slow-played-zunes** [comment]
windows high-contrast themes are decently customizable, so that is a simper solution.
